crates/control/src/role_assignment.rs

```diff
@@ -367,6 +371,7 @@ fn process_role_state_machine(
                 player_number,
                 cycle_start_time,
                 game_controller_state,
+                optional_roles,
```

Comment:
This function takes many arguments. You should consider using a struct with descriptive names. But I'm also fine with the current approach. Feel free to discuss

Reply:
Do you see any advantage of wrapping the arguments in a struct, other than getting rid of the `allow` ?
Or do you have a suggestion for more descriptive names ?

Currently my preference would be to leave it as is.